app/services/articles/feeds/large_forem_experimental.rb

```diff
@@ -126,26 +126,74 @@ def score_comments(article)
             .limited_column_select.includes(top_comments: :user).order(published_at: :desc).limit(rand(15..80))
           hot_stories = hot_stories.to_a + new_stories.to_a
+        else
+          hot_stories = Article.published.limited_column_select
+            .page(@page).per(@number_of_articles)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+          featured_story = hot_stories.where.not(main_image: nil).first
         end
         [featured_story, hot_stories.to_a]
       end
 
       private
 
+      # Disable complexity cop to allow for variant-driven method
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def experimental_hot_story_grab
+        test_variant = @user ? field_test(:feed_top_articles_query, participant: @user) : "base"
+        case test_variant
+        when "base_with_more_articles" # equivalent to current base but with higher "number of articles"
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .page(@page).per(75)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+        when "only_followed_tags" # equivalent to base but only on tags user follows (if user follows enough)
+          followed_tags = @user.cached_followed_tag_names
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .page(@page).per(@number_of_articles)
+            .where("score >= ? OR featured = ?", SiteConfig.home_feed_minimum_score, true)
+            .order(hotness_score: :desc)
+          # We only want to limit the posts to tagged_with if the participant follows enough tags.
+          articles = articles.tagged_with(followed_tags, any: true) if followed_tags.size > 4
+        when "top_articles_since_last_pageview_3_days_max" # Top articles since last page view (max 3 days)
+          start_time = [(@user.page_views.last&.created_at || Time.current) - 12.hours, 3.days.ago].max
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .where("published_at > ?", start_time)
+            .page(@page).per(@number_of_articles)
+            .order(score: :desc)
+        when "top_articles_since_last_pageview_7_days_max" # Top articles since last page view (max 3 days)
+          start_time = [(@user.page_views.last&.created_at || Time.current) - 12.hours, 7.days.ago].max
+          articles = Article.published.limited_column_select.includes(top_comments: :user)
+            .where("published_at > ?", start_time)
+            .page(@page).per(@number_of_articles)
+            .order(score: :desc)
+        when "combination_only_tags_followed_and_top_max_7_days" # Top articles since last page view (max 3 days)
+          start_time = [(@user.page_views.last&.created_at || Time.current) - 12.hours, 7.days.ago].max
```

Comment:
**Question/Suggestion:** I may be way off with this train of thought, but based on this line (that exists in a few of the variants)

```ruby
[(@user.page_views.last&.created_at || Time.current) - 12.hours, 7.days.ago].max
```

These are the two options:
1. If the user has no page views then we're choosing a `start_time` of 12 hours ago (`Time.current - 12.hours` which will always be greater than `7.days.ago`)
2. Otherwise we're choosing a `start_time` of the max of `@user.page_views.last&.created_at - 12.hours` or `7.days.ago`

This means that the newer users will get hot articles from that small window. 

This alternative changes that for option 1 (newer users):

```ruby
[(@user.page_views.last&.created_at || 7.days.ago) - 12.hours, 7.days.ago].max
```

This way the two options now are:
1. If the user has no page views always go with the max range (because the `7.days.ago - 12.hours` will not be chosen over `7.days.ago`)
2. Same as before (max of `@user.page_views.last&.created_at - 12.hours` or `7.days.ago`)

This only affects users that have no page views and I'm not sure if it would have a noticeable impact, so feel free to disregard this in case it doesn't make much sense

Reply:
Yeah that makes sense!